lib/rouge/lexers/powershell.rb

```diff
@@ -652,28 +97,47 @@ class Powershell < Shell
         rule %r/\s+/, Text
         rule %r/\$?"/, Str::Double, :double_quotes
         rule %r/\$'/, Str::Single, :ansi_string
-
         rule %r/'/, Str::Single, :single_quotes
-
         rule %r/\*/, Keyword
-
         rule %r/;/, Text
         rule %r/[^=\*\s{}()$"'`<]+/, Text
         rule %r/\d+(?= |\Z)/, Num
         rule %r/</, Text
         mixin :interp
       end
 
+      state :hashtable do 
+        rule %r/\w/, Name::Variable
+        rule %r/}/, Operator, :pop!
+        mixin :root
+      end
+
+      state :multiline do
+        rule %r/\.(#{MULTILINEKEYWORDS})/i, Keyword::Pseudo
+        rule %r/[\w,\d,\s,\.,\-,\,:\/,{,},<,>"*]/, Comment
+        rule %r/#>/, Comment, :pop!
+        mixin :root
+      end
+
+      state :heredoc do
+        rule %r/"@/, Operator, :pop!
+        rule %r/[^$\n]+/, Str::Heredoc
+        rule %r/[$]/, Str::Heredoc
+        mixin :data
```

Comment:
Same question here about the mixin.

Reply:
Same followup question - not sure what the best approach is.

Reply:
See aaroneg#1.